services/src/northbound/src/main/java/org/openkilda/northbound/service/impl/FlowServiceImpl.java

```diff
@@ -561,20 +566,25 @@ public String toString() {
                 logger.debug("FlowEntry: {}", switchRule);
                 SimpleSwitchRule rule = new SimpleSwitchRule();
                 rule.switchId = rules.getSwitchId();
-                rule.cookie = ""+switchRule.getCookie();
-                rule.inPort = switchRule.getMatch().getInPort();
-                rule.inVlan = switchRule.getMatch().getVlanVid();
-                if (switchRule.getInstructions() != null){
+                rule.cookie = switchRule.getCookie();
+                rule.inPort = NumberUtils.toInt(switchRule.getMatch().getInPort());
+                rule.inVlan = NumberUtils.toInt(switchRule.getMatch().getVlanVid());
+                if (switchRule.getInstructions() != null) {
                     // TODO: What is the right way to get OUT VLAN and OUT PORT?  How does it vary?
                     if (switchRule.getInstructions().getApplyActions() != null) {
                         // The outVlan could be empty. If it is, then pop is?
-                        rule.outVlan = switchRule.getInstructions().getApplyActions().getPushVlan();
-                        if (rule.outVlan != null && rule.outVlan.equals("0x8100")){
-                            rule.outVlan = switchRule.getInstructions().getApplyActions().getFieldAction().getFieldValue();
-                        }
-                        // Is getFlowOutput() the right method?
-                        rule.outPort = switchRule.getInstructions().getApplyActions().getFlowOutput();
+                        FlowApplyActions applyActions = switchRule.getInstructions().getApplyActions();
+                        rule.outVlan = Optional.ofNullable(applyActions.getFieldAction())
+                                .filter(action -> "vlan_vid".equals(action.getFieldName()))
+                                .map(FlowSetFieldAction::getFieldValue)
+                                .map(NumberUtils::toInt)
+                                .orElse(NumberUtils.INTEGER_ZERO);
```

Comment:
Same as above: INTEGER_ZERO explains no more than just 0. Can we have ZERO_VLAN instead?